op-node/rollup/driver/state.go

```diff
@@ -330,14 +334,17 @@ func (s *state) eventLoop() {
 			s.snapshot("New unsafe payload")
 			s.log.Info("Optimistically queueing unsafe L2 execution payload", "id", payload.ID())
 			s.derivation.AddUnsafePayload(payload)
+			s.metrics.UnsafePayloadsTotal.Inc()
 			reqStep()
 
 		case newL1Head := <-s.l1Heads:
 			s.log.Info("new l1 Head")
 			s.snapshot("New L1 Head")
 			s.handleNewL1Block(newL1Head)
+			s.metrics.SetHead("l1", newL1Head.Number)
 			reqStep() // a new L1 head may mean we have the data to not get an EOF again.
 		case <-stepReqCh:
+			s.metrics.SetDerivationIdle(false)
 			s.idleDerivation = false
```

Comment:
would be nice to refactor this into a method that changes the idle state, and only updates metrics if the new state is not the same as the old state.

Reply:
Prom metrics are super efficient. Under the hood this is just an atomic integer set, so I think adding the logic to conditionally set the metric on change will have a negligible performance impact.